### DISCUSSION TITLE
flattenObjectIds and UUID

### DISCUSSION THREAD

**Discussion by AaronPorts** [Automattic/mongoose#14969] [Q&A] [answered]
flattenObjectIds property converts ObjectID into string. Is there any possibility to extend this behavior to UUIDs?

**Comment by vkarpov15** [comment]
It is possible. Tricky for TypeScript, but doable. What is your use case for converting UUIDs to strings?

**Reply by AaronPorts** [reply]
I'm using UUID for refs and _id. Currently `toJSON` returns buffer for reference and the only way to get string value is to use `{ getters: true }`, what may be redundant since we also get all getters.

**Reply by AbdelrahmanHafez** [reply, marked as answer]
@AaronPorts `flattenUUIDs` is added in 9.2. https://github.com/Automattic/mongoose/pull/15864
* [`Document#toJSON` docs](https://mongoosejs.com/docs/api/document.html#Document.prototype.toJSON())
* [`Document#toObject` docs](https://mongoosejs.com/docs/api/document.html#Document.prototype.toObject())

**Reply by vkarpov15** [reply]
@AaronPorts it sounds like you're using Mongoose 8.x, we also released this feature in 8.23.
